source_control/gitlab_tag.py

```diff
@@ -0,0 +1,268 @@
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: gitlab_tag
+short_description: Creates/deletes tags on Gitlab projects
+description:
+   - When the tag does not exists in Gitlab, it will be created.
+   - When the tag does exists and state=absent, the hook will be deleted.
+   - The project and group should already exists when executing this module.
+version_added: "2.3"
+author: "Werner Dijkerman (@dj-wasabi)"
+requirements:
+    - python-gitlab python module
+options:
+    server_url:
+        description:
+            - Url of Gitlab server, with protocol (http or https).
+        required: true
+    validate_certs:
+        description:
+            - When using https if SSL certificate needs to be verified.
+        required: false
+        default: true
+        aliases:
+            - verify_ssl
+    login_user:
+        description:
+            - Gitlab user name.
+        required: false
+        default: null
+    login_password:
+        description:
+            - Gitlab password for login_user
+        required: false
+        default: null
+    login_token:
+        description:
+            - Gitlab token for logging in.
+        required: false
+        default: null
+    group:
+        description:
+            - The name of the group of which the projects belongs to.
+            - If the project belongs to a user, the username should be used for this.
+        required: true
+        default: null
+    project:
+        description:
+            - The name of the project on which the hook needs to be added or updated.
+        required: true
+        default: null
+    name:
+        description:
+            - The name of the tag you want to create
+        required: true
+        default: null
+    ref:
+        description:
+            - Create tag using commit SHA, another tag name, or branch name
+        required: false
+        default: "master"
+    description:
+        description:
+            - A description of the tag.
+        required: false
+        default: null
+    state:
+        description:
+            - create or delete hooks.
+        required: false
+        default: "present"
+        choices: ["present", "absent"]
+
+'''
+
+EXAMPLES = '''
+- name: "Create tags"
+  local_action:
+    module: gitlab_tag
+    server_url="http://gitlab.dj-wasabi.local"
+    validate_certs=false
+    login_token="WnUzDsxjy8230-Dy_k"
+    group: my_first_group
+    project: my_first_project
+    name: v1.0
+    ref: master
+    description: My First Release
+    state: present
+
+- name: "Delete tag"
```

Comment:
I think the example would be clearer if shorter (ie, description do not make much sense if it will be removed, so maybe it can be removed, maybe the same for ref ?)